### ISSUE TITLE
Demon's Souls [BLES00932] flickering rendering issues in the Tower of Latria (3-2) on certain AMD drivers

### ISSUE THREAD

**Issue by Oschowa** [RPCS3/rpcs3#9718]
## Details

The flames of the braziers in the second area of the Tower of Latria (3-2) flicker in and out of existence when using the vulkan renderer on RADV (LLVM and ACO) or AMDVLK, while it is fine when using the proprietary vulkan driver from amdgpu-pro 20.45. There are validation errors triggered in the scene where the problem occurs I'll attach to this report. The OpenGL renderer on radeonsi has a similar but slightly different problem where the flame disappears completely in certain viewing angles, see the videos.

#### 1. Please attach RPCS3's log.
[RPCS3.log.zip](https://github.com/RPCS3/rpcs3/files/5928988/RPCS3.log.zip)

Vulkan validation log: 
[rpcs3-ds-vali.txt](https://github.com/RPCS3/rpcs3/files/5929038/rpcs3-ds-vali.txt)

used config: 
[config_BLES00932.yml.txt](https://github.com/RPCS3/rpcs3/files/5929027/config_BLES00932.yml.txt)

#### 2. If you describe graphical issue, please provide an RSX capture and a RenderDoc capture that demonstrate it.
RSX capture: https://drive.google.com/file/d/1uLfCqB_aYoPet45uXgw5WIzMJjGHTz9A/view?usp=sharing
renderdoc captured with RADV: https://drive.google.com/file/d/16l4LgaPsVnCrUbyxkNhWZEQ-7POJbZm1/view?usp=sharing

#### 3. Please attach screenshots of your problem.
Video of the problem with the vulkan renderer on RADV *or* AMDVLK: https://drive.google.com/file/d/1jCYfBlkY_MDMMl2vrre7PNoMAgyPrkow/view?usp=sharing

Video of the problem with the OpenGL renderer on radeonsi: https://drive.google.com/file/d/1LY70VZENsrzbc-iqiVxdUokRQlNQLBMH/view?usp=sharing

#### 5. Please provide your system configuration:
* OS Arch Linux
* CPU Ryzen 3700x
* GPU RX580
* Driver version Mesa 20.3.4/git *or* AMDVLK 2021.Q1.2

Since this happens with both RADV and AMDVLK and there are validation errors, i opted to report this here first, but I can also open a Mesa/AMDVLK bugreport about this if you are convinced this is a driver issue.

**Comment by kd-11** [comment]
@Oschowa There are two main validation errors I see here, both of which are known:
1. 1D/2D mismatch - known, but usually doesn't cause any issues. I'll remove 1D support and use 2D for everything as it is not possible to support both correctly in most games.
2. Framebuffer/Sampler aliasing (feedback loop) - long time known issue which is fixed when using strict rendering mode to duplicate framebuffers (input attachments are useless here as some games expect filtering and it just works on PS3)
TLDR - Start by enabling strict rendering mode, most of the times it fixes the minor problems but at such a great cost that it's worth just living with them. This applies to both OpenGL and Vulkan.

**Comment by Oschowa** [comment]
Thanks, I tried enabling strict rendering mode, but it unfortunately doesn't fix the rendering issues.
FWIW, I've also tried going back to a ~2 month old build (v0.0.13-11388-e321765c), but the issue is still present.

**Comment by kd-11** [comment]
Still an issue?

**Comment by Oschowa** [comment]
I can't reproduce this anymore on recent rpcs3 and RADV on a 6700xt. I'll also test on a Polaris gpu tomorrow and close if it's fine there as well.

**Comment by Oschowa** [comment]
It's also fine on Polaris, closing.
